signal_windows.go

```diff
@@ -0,0 +1,41 @@
+// +build windows
+
+package prompt
+
+import (
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+func handleSignals(in ConsoleParser, exitCh chan int, winSizeCh chan *WinSize) {
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(
+		sigCh,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT,
+	)
+
+	for {
+		s := <-sigCh
+		switch s {
+		case syscall.SIGINT: // kill -SIGINT XXXX or Ctrl+c
+			log.Println("[SIGNAL] Catch SIGINT")
+			exitCh <- 0
+
+		case syscall.SIGTERM: // kill -SIGTERM XXXX
+			log.Println("[SIGNAL] Catch SIGTERM")
+			exitCh <- 1
+
+		case syscall.SIGQUIT: // kill -SIGQUIT XXXX
+			log.Println("[SIGNAL] Catch SIGQUIT")
+			exitCh <- 0
+
+		default:
+			time.Sleep(10 * time.Millisecond)
```

Comment:
Ok, not trying to Bombard maybe I am still missing gap for go best practices. Why the sleep here?

Reply:
ditto